File: client-app/convert-icon.py
#!/usr/bin/env python3
"""
Convert min-logo.png to icon.ico for the EclipAC application
"""

import os
import sys

def convert_png_to_ico():
    """Convert PNG to ICO using PIL"""
    try:
        from PIL import Image
    except ImportError:
        print("❌ PIL not installed. Installing...")
        os.system("pip install pillow")
        from PIL import Image
    
    input_path = os.path.join(os.path.dirname(__file__), "assets", "min-logo.png")
    output_path = os.path.join(os.path.dirname(__file__), "assets", "icon.ico")
    
    if not os.path.exists(input_path):
        print(f"❌ Error: {input_path} not found")
        return False
    
    try:
        # Open the PNG image
        img = Image.open(input_path)
        
        # Convert RGBA to RGB if needed
        if img.mode == 'RGBA':
            # Create a white background
            background = Image.new('RGB', img.size, (0, 0, 0))
            background.paste(img, mask=img.split()[3])
            img = background
        
        # Resize to appropriate icon size
        img = img.resize((256, 256), Image.Resampling.LANCZOS)
        
        # Save as ICO
        img.save(output_path, format='ICO')
        
        print(f"✓ Successfully converted!")
        print(f"  Input:  {input_path}")
        print(f"  Output: {output_path}")
        print(f"  Size:   256x256")
        return True
    except Exception as e:
        print(f"❌ Conversion error: {e}")
        return False

if __name__ == "__main__":
    success = convert_png_to_ico()
    sys.exit(0 if success else 1)

File: client-app/convert-icon.js
#!/usr/bin/env node

/**
 * Convert PNG to ICO using Node.js
 * Uses the 'to-ico' package
 */

const fs = require('fs');
const path = require('path');
const { exec } = require('child_process');

const inputPath = path.join(__dirname, 'assets', 'min-logo.png');
const outputPath = path.join(__dirname, 'assets', 'icon.ico');

console.log('🔄 Converting logo to icon format...');
console.log(`   Input:  ${inputPath}`);
console.log(`   Output: ${outputPath}`);

// Check if PNG exists
if (!fs.existsSync(inputPath)) {
  console.error(`❌ Error: ${inputPath} not found`);
  process.exit(1);
}

// Install to-ico package
console.log('📦 Installing conversion tool...');
exec('npm install to-ico', { cwd: __dirname }, (error) => {
  if (error) {
    console.error('❌ Failed to install to-ico:', error.message);
    console.log('\n📝 Alternative: Convert manually at https://icoconvert.com/');
    process.exit(1);
  }

  console.log('✓ Tool installed');

  // Convert the image
  try {
    const toIco = require('to-ico');
    const sharp = require('sharp');

    (async () => {
      try {
        // Read and resize the PNG
        const buffer = await sharp(inputPath)
          .resize(256, 256, {
            fit: 'contain',
            background: { r: 0, g: 0, b: 0, alpha: 0 }
          })
          .png()
          .toBuffer();

        // Convert to ICO
        const icoBuffer = await toIco([buffer]);

        // Write ICO file
        fs.writeFileSync(outputPath, icoBuffer);

        console.log('✓ Successfully converted!');
        console.log(`✓ Icon size: 256x256`);
        process.exit(0);
      } catch (err) {
        console.error('❌ Conversion failed:', err.message);
        process.exit(1);
      }
    })();
  } catch (err) {
    console.error('❌ Error:', err.message);
    process.exit(1);
  }
});
